server/controllers/finance/feeCenter.js

```diff
@@ -0,0 +1,263 @@
+/**
+* The /fee_centers HTTP API endpoint
+*
+* @module finance/feeCenters
+*
+* @description This controller is responsible for implementing all crud and others custom request
+* on the fee_center table through the `/fee_centers` endpoint.
+*
+* @requires lib/db
+**/
+
+
+var db = require('../../lib/db');
+var NotFound = require('../../lib/errors/NotFound');
+
+/**
+* Returns an array of fee centers
+*
+* @param {object} req The express request object
+* @param {object} res The express response object
+* @param {object} next The express object to pass the control to the next middleware
+*
+* @example
+* // GET /fee_centers : Get list of fee_centers
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.list(req, res, next);
+*/
+
+function list (req, res, next) {
+  'use strict';
+
+  var sql =
+    'SELECT fc.id, fc.label, fc.is_cost FROM fee_center AS fc ';
+  var filter = '', ordering = 'ORDER BY fc.label;';
+
+  if (req.query.full === '1') {
+    sql =
+      `SELECT fc.id, fc.label, fc.project_id, fc.is_cost, fc.note, fc.is_principal, p.name, p.abbr, p.enterprise_id, p.zs_id
+      FROM fee_center AS fc JOIN project AS p ON fc.project_id = p.id `;
+  }
+
+
+  if(req.query.available === '1') {
+    filter += ' fc.id NOT IN (SELECT s.cc_id FROM service AS s WHERE NOT ISNULL(s.cc_id)) AND fc.id NOT IN (SELECT s.pc_id FROM service AS s WHERE NOT ISNULL(s.pc_id)) ';
+  }
+
+  if(req.query.is_cost){
+    if(filter !== ''){
+      filter += ' AND ';
+    }
+    filter += `fc.is_cost = ${req.query.is_cost} `;
+  }
+
+  if(req.query.is_principal){
+    if(filter !== ''){
+      filter += ' AND ';
+    }
+    filter += `fc.is_principal = ${req.query.is_cost} `;
+  }
+
+  sql = filter === '' ? sql + ordering : sql + 'WHERE ' + filter + ordering;
```

Comment:
If you used `db.exec(sql, [params])`, you could write your code so that you didn't have to check if `filter === ''` all the time.